app/controllers/find/v2/subjects_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Find
+  module V2
+    class SubjectsController < Find::ApplicationController
+      def primary
+        @form = Find::V2::Subjects::PrimarySubjectsForm.new(subject_params)
+        @primary_subject_options = Subject.primary
+      end
+
+      def submit
+        @form = Find::V2::Subjects::PrimarySubjectsForm.new(subject_params)
+
+        if @form.save
+          redirect_to find_v2_results_path(search_query_params)
+        else
+          @primary_subject_options = Subject.primary
+          render :primary
+        end
+      end
+
+      private
+
+      def subject_params
+        params.fetch(:find_v2_subjects_primary_subjects_form, {}).permit(subjects: [])
+      end
+
+      def search_query_params
+        {
+          subjects: @form.subjects,
+          university_degree_status: true,
```

Comment:
This is an old parameter for showing only postgraduate. We will only show postgraduate after search for primary?

Reply:
But we won't have the questions "Yes, I have a degree or am studying for one" on the pre-filtering. Sorry my confusion. And this will filter for only postgraduate courses. Also this is the old parameter. If we want to have only postgraduate, you need to use the minimum degree required as 1:1. 

My question would be the reasoning behing setting those for the user when the user didn't request that particularly. 

If I am the user and ask to browse Primary courses I want to see all Primary courses regarless of degree type or visa?

Let's talk on Slack.